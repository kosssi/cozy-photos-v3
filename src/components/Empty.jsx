import styles from '../styles/empty'

import React from 'react'
import { translate } from '../lib/I18n'

export const Empty = ({ t, emptyType }) => {
  return (
    <div>
      {emptyType === 'photos' &&
        <div className={styles['pho-empty']}>
          <h2>{ t('Empty.photos_title') }</h2>
          <p>{ t('Empty.photos_text')}</p>
        </div>
      }
      {emptyType === 'albums' &&
        <div className={styles['pho-empty']}>
          <h2>{ t('Empty.albums_title') }</h2>
          <p>{ t('Empty.albums_text')}</p>
        </div>
      }
    </div>
  )
}

export default translate()(Empty)
