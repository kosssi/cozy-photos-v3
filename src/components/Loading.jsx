import styles from '../styles/loading'

import React from 'react'
import { translate } from '../lib/I18n'

export const Loading = ({ t, loadingType }) => {
  return (
    <div className={styles['pho-loading']}>
      <p>{t(`Loading.${loadingType}`)}</p>
    </div>
  )
}

export default translate()(Loading)
