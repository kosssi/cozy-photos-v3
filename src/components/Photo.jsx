import styles from '../styles/photo'

import React from 'react'

const STACK_FILES_DOWNLOAD_PATH = 'http://cozy.local:8080/files/download'

const Photo = ({ photo }) => {
  return (
    <img
      className={styles['pho-photo']}
      src={`${STACK_FILES_DOWNLOAD_PATH}/${photo._id}`}
    />
  )
}

export default Photo
