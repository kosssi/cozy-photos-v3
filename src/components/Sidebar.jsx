import styles from '../styles/sidebar'

import React from 'react'

import Nav from './Nav'

const Sidebar = () => (
  <aside className={styles['coz-sidebar']}>
    <Nav />
  </aside>
)

export default Sidebar
