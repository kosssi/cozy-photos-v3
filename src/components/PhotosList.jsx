import styles from '../styles/photosList'

import React from 'react'
import { translate } from '../lib/I18n'

import Empty from '../components/Empty'
import Loading from '../components/Loading'
import Photo from '../components/Photo'

const getPhotosSectionsByMonth = (f, photos) => {
  let sections = {}
  photos.map(p => {
    /* istanbul ignore else */
    if (!sections.hasOwnProperty(f(p.created_at, 'MMMM YYYY'))) {
      sections[f(p.created_at, 'MMMM YYYY')] = []
    }
    sections[f(p.created_at, 'MMMM YYYY')].push(p)
  })
  return sections
}

export const PhotosList = ({ t, f, photos, isIndexing, isFetching, isWorking, isFirstFetch }) => {
  if (isIndexing) {
    return <Loading loadingType='photos_indexing' />
  }
  if (isFetching || isFirstFetch) {
    return <Loading loadingType='photos_fetching' />
  }
  if (isWorking) {
    return <Loading loadingType='photos_upload' />
  }
  const sections = getPhotosSectionsByMonth(f, photos)
  return (
    <div role='contentinfo'>
      {Object.keys(sections).map(sectionName => {
        return (
          <div class={styles['pho-section']} key={sectionName}>
            <h3>{sectionName}</h3>
            {sections[sectionName].map(photo => {
              return <Photo photo={photo} key={photo._id} />
            })}
          </div>
        )
      })}
      {photos.length === 0 && <Empty emptyType='photos' />}
    </div>
  )
}

export default translate()(PhotosList)
