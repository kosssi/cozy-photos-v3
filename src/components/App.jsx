import styles from '../styles/app'

import React from 'react'
import { translate } from '../lib/I18n'
import classNames from 'classnames'

import Sidebar from './Sidebar'
import Alerter from './Alerter'

export const App = ({ t, children }) => (
  <div className={classNames(styles['pho-wrapper'], styles['coz-sticky'])}>

    <Sidebar />

    <main className={styles['pho-content']}>
      { children }
    </main>
    <Alerter t={t} />
  </div>
)

export default translate()(App)
