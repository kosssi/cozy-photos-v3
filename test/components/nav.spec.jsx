'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import { mockT } from '../lib/I18n'
import { Nav } from '../../src/components/Nav'

describe('Nav component', () => {
  it('should be rendered correctly', () => {
    const component = shallow(
      <Nav t={mockT} />
    ).node
    expect(component).toMatchSnapshot()
  })
})
