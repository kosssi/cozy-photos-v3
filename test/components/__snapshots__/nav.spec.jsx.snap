exports[`Nav component should be rendered correctly 1`] = `
<nav>
  <ul
    className="coz-nav">
    <li
      className="coz-nav-item">
      <Link
        activeClassName="active"
        className="pho-cat-photos"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/photos">
        Photos
      </Link>
    </li>
    <li
      className="coz-nav-item">
      <Link
        activeClassName="active"
        className="pho-cat-albums"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/albums">
        Albums
      </Link>
    </li>
    <li
      className="coz-nav-item">
      <Link
        activeClassName="active"
        className="pho-cat-shared"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/shared">
        Shared by me
      </Link>
    </li>
    <li
      className="coz-nav-item">
      <Link
        activeClassName="active"
        className="pho-cat-trash"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/trash">
        Trash
      </Link>
    </li>
  </ul>
</nav>
`;
